Color picker netcore 31

snickler opened the pull request:
## Summary of the Pull Request

Conversion of the Color Picker module from .NET Framework to .NET Core 3.1.

## PR Checklist
* [X] Applies to #776
* [X] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA
* [X] Tests added/passed
* [ ] Requires documentation to be updated
* [X] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

_What does this include?_

## Validation Steps Performed

The Unit Tests for the Color Picker run successfully, the module itself can be run properly in both a standalone instance and within the runner. Colors are properly detected and selected. The key combination works properly.

davidegiacometti commented:
Hi @snickler 
Installer build failed.
Need to fix the list of files deployed for ColorPickerUI:
https://github.com/microsoft/PowerToys/blob/b1c0c5a03eddfcc794116fd01860f32ca902d5e5/installer/PowerToysSetup/Product.wxs#L580
Also after porting to .NET Core 3.1 there should be also `ColorPicker.dll`, `ColorPicker.deps.json` and `ColorPicker.runtimeconfig.json`.
There is already a `ColorPicker.dll` created by the C++ program and the projects are deployed in the same directory when installed.

snickler commented:
> Hi @snickler
> Installer build failed.
> Need to fix the list of files deployed for ColorPickerUI:
> https://github.com/microsoft/PowerToys/blob/b1c0c5a03eddfcc794116fd01860f32ca902d5e5/installer/PowerToysSetup/Product.wxs#L580
> 
> 
> Also after porting to .NET Core 3.1 there should be also `ColorPicker.dll`, `ColorPicker.deps.json` and `ColorPicker.runtimeconfig.json`.
> There is already a `ColorPicker.dll` created by the C++ program and the projects are deployed in the same directory when installed.

Thanks! I'll get that changed. 

@davidegiacometti - Can you also take a look at my branch and see why the ColorPicker crashes the runner, but works when you run the ColorPicker by itself?

davidegiacometti commented:
Can give a check tomorrow!
I think the problem here is that there are 2 projects (C# and C++) that creates a ColorPicker.dll since now is .NET Core.
Rename this one to ColorPickerUI (ColorPickerUI.dll) could be a solution. @martinchrzan any thoughts?

martinchrzan commented:
Why would there be 2 ColorPicker.dll files? C++ project is a library and should produce .dll but ColorPickerUI project (C#) should be published to produce .exe file and be framework dependend (as we ensure users have .net core installed). It is not a platform independent project that is used as a .dll library.

davidegiacometti commented:
In .NET Core the output is a dll even for WPF and console application. We also have the exe but the dll is required.

snickler commented:
> Can give a check tomorrow!
> I think the problem here is that there are 2 projects (C# and C++) that creates a ColorPicker.dll since now is .NET Core.
> Rename this one to ColorPickerUI (ColorPickerUI.dll) could be a solution. @martinchrzan any thoughts?

That's exactly what it was. After I renamed the assembly to ColorPickerUI and updated the path to the exe within the module, I'm able to run it properly in the runner. Now to fix the installer

martinchrzan commented on src/modules/colorPicker/ColorPickerUI/Controls/ColorPickerControl.xaml.cs:
```diff
@@ -393,7 +393,7 @@ private void SetColorFromTextBoxes(System.Drawing.Color color)
         private static string ColorToHex(Color color)
         {
-            return "#" + BitConverter.ToString(new byte[] { color.R, color.G, color.B }).Replace("-", string.Empty);
+            return "#" + BitConverter.ToString(new byte[] { color.R, color.G, color.B }).Replace("-", string.Empty, StringComparison.CurrentCulture);
```
InvariantCulture

martinchrzan commented:
Please update pipeline.user.windows.yml too

snickler commented:
> Please update pipeline.user.windows.yml too

Done via the latest commit. Thanks for bringing up the yml pipeline. I had no idea about that :)

crutkas commented:
@enricogior, is renaming the exe better or the module dll?

enricogior commented:
@crutkas 
now the project name and the project binaries are consistent, that seems a good choice.
